lib/serializer.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const Serializer = require('parse5/lib/serializer');
+
+module.exports = class CustomSerializer extends Serializer {
+
+    constructor(node, options) {
+        super(node, options);
+        this.slotMap = options.slotMap;
+        this.handleTags = options.handleTags;
+        this.pipeTags = options.pipeTags;
+        this.isPipeInserted = false;
+        this.lastChildInserted = false;
+        this.serializedList = [];
+        this._serializeNode = this._serializeNode.bind(this);
+    }
+
+    pushBuffer() {
+        if (this.html !== '') {
+            this.serializedList.push(new Buffer(this.html));
+            this.html = '';
+        }
+    }
+
+    serialize() {
+        this._serializeChildNodes(this.startNode);
+        this.pushBuffer();
+        return this.serializedList;
+    }
+
+    _isPipeNode(node) {
+        return this.pipeTags.indexOf(node.name) !== -1;
+    }
+
+    _isSlotNode(node) {
+        if (node.name === 'slot') {
+            return true;
+        }
+        const attribs = node.attribs;
+        if (!attribs) {
+            return false;
+        }
+        return node.name === 'script' && attribs.type === 'slot';
```

Comment:
not sure, this improves readability a lot. but this could also be written as:
`return node.name === 'script' && attribs && attribs.type === 'slot'`